lib/api.js

```diff
@@ -218,7 +218,7 @@ export default class Api extends Emittery {
 			}
 
 			runStatus.on('stateChange', record => {
-				if (record.testFile && !timedOutWorkerFiles.has(record.testFile)) {
+				if (record.testFile && !timedOutWorkerFiles.has(record.testFile) && record.type !== 'worker-stderr' && record.type !== 'worker-stdout') {
```

Comment:
I'd personally prefer to make this opt-in so only assertions/test state changes reset the timeout (so we hew closer to the documentation). 
But:
a) I don't know the complete list of record type's here that would apply
b) it's a bigger change from current behavior than I'm comfortable making without feedback

Reply:
I think this change is correct. The timeout here is a guard against test workers that are no longer making progress. It's different from `t.timeout()` which limits how long an individual test can take.

Essentially with this change, we no longer count stdout/stderr as "making progress".